addons/sms/wizard/sms_composer.py

```diff
@@ -320,10 +321,11 @@ def _prepare_mass_sms_values(self, records):
                 'state': state,
                 'failure_type': failure_type,
             }
-        return result
+        return (result if not self.composition_mode == 'mass' or mass_include_canceled
+                else {res_id: values for res_id, values in result.items() if values.get('state') != 'canceled'})
 
     def _prepare_mass_sms(self, records, sms_record_values):
-        sms_create_vals = [sms_record_values[record.id] for record in records]
+        sms_create_vals = [sms_record_values[record.id] for record in records if record.id in sms_record_values]
```

Comment:
Maybe a bit out of scope but it's weird because as far as I can tell this should work fine with just `list(sms_record_values.values())`, because records is always the same as what we pass in sms_record_values.

I want to say the iteration was never needed. Can you try removing it to see if any test actually fails?